test/DynamoCoreTests/Graph/Nodes/NodeModelTests.cs

```diff
@@ -78,6 +78,53 @@ public void SelectNeighborsTest()
             Assert.AreEqual(2, countAfter);
         }
 
+        [Test]
+        [Category("UnitTests")]
+        public void SelectUpstreamNeighborsTest()
```

Comment:
lets use AAA (Arrange-Act-Assert) pattern for tests https://docs.microsoft.com/en-us/visualstudio/test/unit-test-basics?view=vs-2019#write-your-tests